include/zephyr/drivers/adc/current_sense_amplifier.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_ADC_CURRENT_SENSE_AMPLIFIER_H_
+#define ZEPHYR_INCLUDE_DRIVERS_ADC_CURRENT_SENSE_AMPLIFIER_H_
+
+#include <zephyr/drivers/adc.h>
+
+struct current_sense_amplifier_dt_spec {
+	const struct adc_dt_spec port;
+	uint32_t sense_micro_ohms;
+	uint32_t sense_gain_mult;
+	uint32_t sense_gain_div;
+};
+
+/**
+ * @brief Get current sensor information from devicetree.
+ *
+ * This returns a static initializer for a @p current_sense_amplifier_dt_spec structure
+ * given a devicetree node.
+ *
+ * @param node_id Devicetree node identifier.
+ *
+ * @return Static initializer for an current_sense_amplifier_dt_spec structure.
+ */
+#define CURRENT_SENSE_AMPLIFIER_DT_SPEC_GET(node_id)                                               \
+	{                                                                                          \
+		.port = ADC_DT_SPEC_GET(node_id),                                                  \
+		.sense_micro_ohms = DT_PROP(node_id, sense_resistor_micro_ohms),                   \
+		.sense_gain_mult = DT_PROP(node_id, sense_gain_mult),                              \
+		.sense_gain_div = DT_PROP(node_id, sense_gain_div),                                \
+	}
+
+/**
+ * @brief Calculates the actual voltage from the measured voltage
```

Comment:
s/voltage/amperage/

Reply:
I've opened a new PR to fix these errors #59892.